### ISSUE TITLE
Unexpected behaviour when forceable and public are used in the same project

### ISSUE THREAD

**Issue by lucasasselli** [verilator/verilator#3955]
On the git master version. Using both `/* verilator public */` and `/* verilator forceable */` results in the force control signals to be created at the module level as `__PVT__` instead of public_flat.

```
module foo(
    input a /* verilator forceable */,
    input b,
    output c
    );

logic t1;

assign t1 = a & b;
assign c = t1;

endmodule

module bar(
    input x,
    input y,
    output z
    );

logic  t2 /* verilator public */;
logic t3;

foo u_foo(x, y, t3);

assign t2 = t3 | x;
assign z = t2;

endmodule
```

```
verilator --cc --top bar top.sv
```

![image](https://user-images.githubusercontent.com/11899876/218278915-7dff9d15-4916-40cd-95e0-4ad9e026118e.png)

![image](https://user-images.githubusercontent.com/11899876/218278945-9ff435c0-d785-4d99-82b0-fcd417315bbe.png)

**Comment by wsnyder** [comment]
Perhaps you could contribute a pull request with a test and fix - presumably some spot maybe in V3Name needs a !isPublic().

**Comment by lucasasselli** [comment]
I will look into it! Is the expectation that the control signals are generated as flat, correct?

**Comment by wsnyder** [comment]
> I will look into it! Is the expectation that the control signals are generated as flat, correct?

Yes, currently V3Force works post flattening (after V3Scope and before V3Descope)

**Comment by wsnyder** [comment]
Sweeping old bugs... the behavior desired I think can be achieved with:

```
    input a  /* verilator forceable */ /* verilator public_flat */,
```

Will commit a tests to make sure this keeps working.
